sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/QuickPulsePingSender.java

```diff
@@ -107,10 +112,16 @@ QuickPulseHeaderInfo ping(String redirectedEndpoint) {
 
             if (networkHelper.isSuccess(response)) {
                 QuickPulseHeaderInfo quickPulseHeaderInfo = networkHelper.getQuickPulseHeaderInfo(response);
+                QuickPulseMetricReceiver.setQuickPulseHeaderInfo(quickPulseHeaderInfo);
                 switch (quickPulseHeaderInfo.getQuickPulseStatus()) {
                     case QP_IS_OFF:
                     case QP_IS_ON:
                         lastValidTransmission = sendTime;
+                        String etagValue = networkHelper.getEtagHeaderValue(response);
+                        if (etagValue != null) {
+                            ConcurrentHashMap<String, ArrayList<QuickPulseConfiguration.OpenTelMetricInfo>> otelMetrics = quickPulseConfiguration.parseMetrics(response);
```

Comment:
Just want to reiterate that it is possible for a response to contained "DerivedMetricInfo" that are completely unrelated to the "Metric" telemetry type (think, Request/Dependency/Exception/Trace/Event). These methods (parseMetrics & updateConfig) should be written in such a way that if a DerivedMetricInfo of a different telemetry type were to be present in the config, that your current code should not fail or mix up telemetry types when tracking otel metrics the customer asked for. I noted this on the QuickPulseConfiguration class as well.

Reply:
Updated code should not fail or mix up telemetry names. When the collector goes through collected metrics and cross references them with the parsed derived metrics, it also checks if the telemetry type is "Metric"